### ISSUE TITLE
`truffle migrate` on private permissioned Quorum network returns nothing to console

### ISSUE THREAD

**Issue by saitam1** [trufflesuite/truffle#1622]
- [x] I've asked for help in the [Truffle Gitter](http://gitter.im/Consensys/truffle) before filing this issue.

---------------------------

## Issue
In broad strokes, I am running a private permissioned Quorum network and would like to use Truffle for deployment. After the step `truffle migrate --network networkname` truffle seems to do some work for about 10 seconds, then stops the process and returns **nothing** to the console!
Here is how I configured `truffle-config`:
```
private: {
  provider: () => new HDWalletProvider(myPrivateKey, `http://ip_address_to_network`),
  port: 22000,
  network_id: "*",
  from: "0x....",
  gasPrice: 0,
  gas: 4500000
}
```

## Steps to Reproduce

Setup a Quorum network with `raft` consensus mechanism and run truffle for contract deployment 

## Expected Behavior

I would like truffle to deploy a contract or at least **to have an error message**

## Actual Results

truffle returns **nothing** to the console

## Environment

* Operating System: Ubuntu 16.04
* Ethereum client (geth --version): 1.8.12-stable 
* Truffle version (`truffle version`): v5.0.1 (core: 5.0.1)
* node version (`node --version`): v10.14.2
* npm version (`npm --version`): 6.5.0

**Comment by fixanoid** [comment]
@saitam1 I've been playing with truffle v5 and I'm seeing similar issue. I am not seeing it with v4.x tho. Do you mind downgrading to confirm while I'm trying to find the origin? Thanks

**Comment by saitam1** [comment]
@fixanoid
The following versions (`4.1.11` - `4.1.14`) seem to have some issue with `truffle init`, they alle give the same error: `Truffle Box at URL https://github.com/truffle-box/bare-box.git doesn't exist. If you believe this is an error, please contact Truffle support.`

Version `4.1.15` however seem to work better with Quorum and, funnily enough, only gets stuck every second try if you do `truffle migrate --reset`. Without `--reset` it works even better(!), but this is where a new problem comes into play. Truffle runs the migration and successfully deploys `contract 1`, `contract 2` and then at `contract 3`: 
> Error: The contract code couldn't be stored, please check your gas amount.

Locally with ganache this contract uses the following amount of gas:
`gas used:            1063899`

And on Quorum I use `gas: 8000000`, `gasPrice: 0`, so I don't know what the issue here is :-/

**Comment by fixanoid** [comment]
@saitam1 thanks for confirming this for me.

The contract code error at contract 3 should not be fatal, contract code storage in Ethereum / Quorum is limited. If the txn was successful then your contract is on the chain.

**Comment by kyriediculous** [comment]
Same issue on Geth PoA. 

Getting a bunch of issues after upgrading to truffle 5.0.1 that I did not have before. 

without `--reset` my contracts compile and then I get an error.

With `--reset `my contracts compile and then it just hangs. 

There are no gas issues or errors in my contracts. One thing I did do is force the compiler down to 0.4.23 because some lunatic in Solidity thought it would be a good idea to make `reference` a reserved keyword.

```
Error: Returned values aren't valid, did it run Out of Gas?
    at ABICoder.decodeParameters (C:\Users\N.vergauwen\AppData\Roaming\npm\node_modules\truffle\build\webpack:\~\web3-eth-abi\src\index.js:226:1)
    at Contract._decodeMethodReturn (C:\Users\N.vergauwen\AppData\Roaming\npm\node_modules\truffle\build\webpack:\~\web3-eth-contract\src\index.js:465:1)
    at Method.outputFormatter (C:\Users\N.vergauwen\AppData\Roaming\npm\node_modules\truffle\build\webpack:\~\web3-eth-contract\src\index.js:818:1)
    at Method.formatOutput (C:\Users\N.vergauwen\AppData\Roaming\npm\node_modules\truffle\build\webpack:\~\web3-eth-contract\~\web3-core-method\src\index.js:163:1)
    at sendTxCallback (C:\Users\N.vergauwen\AppData\Roaming\npm\node_modules\truffle\build\webpack:\~\web3-eth-contract\~\web3-core-method\src\index.js:473:1)
    at C:\Users\N.vergauwen\AppData\Roaming\npm\node_modules\truffle\build\webpack:\~\web3-core-requestmanager\src\index.js:147:1
    at Object.intermediary [as callback] (C:\Users\N.vergauwen\AppData\Roaming\npm\node_modules\truffle\build\webpack:\~\ganache-core\lib\provider.js:114:1)
    at self.engine.sendAsync (C:\Users\N.vergauwen\AppData\Roaming\npm\node_modules\truffle\build\webpack:\~\ganache-core\lib\provider.js:169:1)
    at C:\Users\N.vergauwen\AppData\Roaming\npm\node_modules\truffle\build\webpack:\~\ganache-core\~\web3-provider-engine\index.js:152:1
    at C:\Users\N.vergauwen\AppData\Roaming\npm\node_modules\truffle\build\webpack:\~\async\internal\once.js:12:1
    at replenish (C:\Users\N.vergauwen\AppData\Roaming\npm\node_modules\truffle\build\webpack:\~\async\internal\eachOfLimit.js:61:1)
    at C:\Users\N.vergauwen\AppData\Roaming\npm\node_modules\truffle\build\webpack:\~\async\internal\eachOfLimit.js:71:1
    at eachLimit (C:\Users\N.vergauwen\AppData\Roaming\npm\node_modules\truffle\build\webpack:\~\async\eachLimit.js:43:1)
    at C:\Users\N.vergauwen\AppData\Roaming\npm\node_modules\truffle\build\webpack:\~\async\internal\doLimit.js:9:1
    at Object.end [as callback] (C:\Users\N.vergauwen\AppData\Roaming\npm\node_modules\truffle\build\webpack:\~\ganache-core\~\web3-provider-engine\index.js:127:1)
    at intermediary (C:\Users\N.vergauwen\AppData\Roaming\npm\node_modules\truffle\build\webpack:\~\ganache-core\lib\statemanager.js:401:1)
    at C:\Users\N.vergauwen\AppData\Roaming\npm\node_modules\truffle\build\webpack:\~\ganache-core\lib\statemanager.js:626:1
    at C:\Users\N.vergauwen\AppData\Roaming\npm\node_modules\truffle\build\webpack:\~\ganache-core\lib\blockchain_double.js:482:1
    at StateManager.proto.revert (C:\Users\N.vergauwen\AppData\Roaming\npm\node_modules\truffle\build\webpack:\~\ganache-core\~\ethereumjs-vm\dist\stateManager.js:269:1)
    at cleanUpAndReturn (C:\Users\N.vergauwen\AppData\Roaming\npm\node_modules\truffle\build\webpack:\~\ganache-core\lib\blockchain_double.js:477:1)
    at C:\Users\N.vergauwen\AppData\Roaming\npm\node_modules\truffle\build\webpack:\~\ganache-core\lib\blockchain_double.js:529:1
    at C:\Users\N.vergauwen\AppData\Roaming\npm\node_modules\truffle\build\webpack:\~\ganache-core\~\ethereumjs-vm\dist\runTx.js:44:1
    at C:\Users\N.vergauwen\AppData\Roaming\npm\node_modules\truffle\build\webpack:\~\async\dist\async.js:3888:1
    at C:\Users\N.vergauwen\AppData\Roaming\npm\node_modules\truffle\build\webpack:\~\async\dist\async.js:473:1
    at replenish (C:\Users\N.vergauwen\AppData\Roaming\npm\node_modules\truffle\build\webpack:\~\async\dist\async.js:1006:1)
    at iterateeCallback (C:\Users\N.vergauwen\AppData\Roaming\npm\node_modules\truffle\build\webpack:\~\async\dist\async.js:995:1)
    at C:\Users\N.vergauwen\AppData\Roaming\npm\node_modules\truffle\build\webpack:\~\async\dist\async.js:969:1
    at C:\Users\N.vergauwen\AppData\Roaming\npm\node_modules\truffle\build\webpack:\~\async\dist\async.js:3885:1
Truffle v5.0.1 (core: 5.0.1)
Node v10.13.0

```

**Comment by edgraaff** [comment]
It would be _really_ useful if it doesn't fail silently and prints caught errors instead. I don't mind getting my hands dirty and do some debugging, but I simply have no idea where to start.

**Comment by dappsar** [comment]
I have the same issue with truffle version 5, but not with truflfe 4.  To do a downgrade is a workaround but doesn´t work if you have sentences that need to be compile with solc version 5 (included in truffle 5)

**Comment by dan-turner** [comment]
Same problem here trying to connect to Kaleido...

**Comment by benjamincburns** [comment]
This will likely be fixed by #1543.

**Comment by seesemichaelj** [comment]
I agree with @benjamincburns's diagnosis. In this case, Truffle is swallowing an error that is very likely to be the same as #1543; I have seen this issue's behavior as well as a symptom of #1543. PR #1806 should alleviate this issue. It's very possible though that #1656 may become an issue for you shortly after however.

**Comment by seesemichaelj** [comment]
We just released https://github.com/trufflesuite/truffle/releases/tag/v5.0.9 which fixed #1543, and it may fix this issue as well. Can you give it a try?

You can [read documentation on the necessary changes to use Quorum here](https://truffleframework.com/docs/truffle/getting-started/working-with-quorum).

**Comment by mosasiru** [comment]
I tried v5.0.19. and confirmed `truffle migrate` showed console messages. Thank you!
However `truffle console` returns nothing to console still.

**Comment by CruzMolina** [comment]
hey @mosasiru , can you share an example repo with a truffle project exhibiting this behavior?

**Comment by mosasiru** [comment]
I'm so sorry, that was my misunderstanding. I found it work well!

**Comment by CruzMolina** [comment]
Closing as I believe this has been resolved as of truffle 5.0.9 via #1543.
